File: Non basic operations/Power/power.py
def power(a: int, n: int):
    count = 1
    p = a

    while count < n:
        p = p * a
        count += 1

    return p


# recursive
def powerRecursive(a: int, n: int):
    if n == 0:
        return 1
    else:
        return a * powerRecursive(a, n - 1)


a = int(input("Base: "))
n = int(input("Exponent: "))

# comparing results
print(f"{a} raised to the power of {n} is {power(a, n)}. (NON RECURSIVE)\n")
print(f"{a} raised to the power of {n} is {powerRecursive(a, n)}. (RECURSIVE)\n")
# result with native python function
print(f"{a} raised to the power of {n} is {pow(a, n)}. (PYTHON)\n")

File: Non basic operations/Power/power.cpp
#include <iostream>
#include <math.h>

int power(int a, int n)
{
    int count = 1;
    int p = a;

    while (count < n)
    {
        p = p * a;
        count += 1;
    }

    return p;
}

int recursivePower(int a, int n)
{
    if (n == 0)
    {
        return 1;
    }
    else
    {
        return a * recursivePower(a, n - 1);
    }
}

int main()
{
    int a, n;
    std::cout << "Base: ";
    std::cin >> a;
    std::cout << "Exponent: ";
    std::cin >> n;

    std::cout << power(a, n) << " (NON RECURSIVE)" << std::endl;
    std::cout << recursivePower(a, n) << " (RECURSIVE)" << std::endl;
    std::cout << pow(a, n) << " (C++)" << std::endl;

    return 0;
}